### ISSUE TITLE
Move all community plugins under ix-plugin-hub?

### ISSUE THREAD

**Issue by jcassette** [ix-plugin-hub/iocage-plugin-index#288]

**Comment by jasker5183** [comment]
Maybe the artifact files too?

**Comment by tprelog** [comment]
I think it's a bad idea. It takes long enough to just get the plugin manifest updated and into a release branch

If I had to wait as long to get the artifact repo updated, plugins would be broke for months at at time, just waiting on something to be merged. -- For example, I submitted an upgrade to the `homeassistant.json` back in November to fix a broken plugin. It was only a few days ago these changes make it to the release branch. Since November I've been installing the missing packages during the plugin's post_install. If I didn't have control over my own artifact repo the plugin would have broke for almost 4 months

**Comment by jcassette** [comment]
Thanks for your reply, I had not thought about this, let me elaborate a bit.

First, the plugins below are under "freenas" (official) but it's not consistent because they are community plugins:

- https://github.com/freenas/iocage-plugin-backuppc.git
- https://github.com/freenas/iocage-plugin-clamav.git
- https://github.com/freenas/iocage-plugin-emby.git
- https://github.com/freenas/iocage-plugin-irssi.git
- https://github.com/freenas/iocage-plugin-jenkins-lts.git
- https://github.com/freenas/iocage-plugin-jenkins.git
- https://github.com/freenas/iocage-plugin-madsonic.git
- https://github.com/freenas/iocage-plugin-qbittorrent.git
- https://github.com/freenas/iocage-plugin-quassel.git
- https://github.com/freenas/iocage-plugin-radarr.git
- https://github.com/freenas/iocage-plugin-sonarr.git
- https://github.com/freenas/iocage-plugin-transmission.git
- https://github.com/freenas/iocage-plugin-weechat.git
- https://github.com/freenas/iocage-plugin-xmrig.git
- https://github.com/freenas/iocage-plugin-zoneminder.git

For other plugins, I understand you point, but what if the plugin owner deletes the repo?

In your case maybe TrueNAS is missing an option to add a custom index, or the artifact should be in the plugin repo (the documentation says it is supported).

Finally, the documentation suggests that the repo should be forked:

> Remember to put a link to your newly created artifact repo in the comments of the pull request. This way a moderator can fork your repo and it can be made available in the community list of plugins.

https://www.truenas.com/docs/core/applications/plugins/createplugin/

**Comment by Baenwort** [comment]
Yes, and those FreeNAS plugin .gits are there because they are all old enough that it was before the TrueNAS acquisition and they tried to follow the ancestor of that guide by doing exactly what it says.

This resulted in the problems talked about above and the solution by the community was to move to what we have now as with the decreasing resources being spent on plug-ins ment it took a LONG time to make any fixes. The ones that are still under /freenas/ are ones that haven't needed much done or are maintained by people who are old enough to remember before but haven't needed to fix. 

I don't think migrating back to the system that failed is any better. We haven't had any instances that I recall in the last 8 years of someone pulling their plugin .git entirely. All the cases where a maintainer has dropped out have had someone else come along, fork it, and start taking care of it.
